text/0072-signer-vault-recovery.md

```diff
@@ -0,0 +1,86 @@
+
+There is a non-trivial amount of work involved in the implementation as well as the introduction of potential security risks to what is otherwise a very secure setup.  
+It is also another secret for users to manage (in addition to the current **Vault Password**).
+
+In the CEP's current state there would need to be a **Secret Phrase** per keypair which could be an inconvenience for users looking to manage multiple keypairs.  
+See the [future possibilities][future-possibilities] section for a potential enhancement.
+
+## Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This implementation for recovery of keypairs is already being used by many entities e.g. [MetaMask][metamask].
+Alternative methods used in non-blockchain services such as **email** or **phone number** may not be as acceptable to a community who value anonymity.
+
+Another alternative would be to actively and strongly encourage users to download their Signer generated keys immediately after creation. There could be an option to 'Skip' but it would be made clear that this is at the user's own risk. Something similar exists in [MetaMask][metamask] but applied to the backup of the Secret Recovery Phrase (since it doesn't support downloading actual key files). 
+
+## Prior art
+
+[prior-art]: #prior-art
+
+MetaMask had some issues with terminology when they referred to it as a "Seed Phrase". It didn't imply the significance of the data and some users shared it with malicious parties who were then able to take control of the user's accounts. They recently opted to rename it a "Secret Recovery Phrase" and I think our terminology of **Secret Phrase** should adequately communicate the sensitive nature of the data.  
+Using mnemonics/phrases to recover keys is common throughout the crypto space as it requires no personal/contact information to perform the recovery.
+
+## Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+<sup>[1]</sup> The length of the phrase is commonly 12 words however it can be set to 24 for added security although I feel that benefit is outweighed by the shorter, more ergononmic 12 word phrase.
+
+As part of this change should we also phase out the ability to download the key files themselves - opting to purely use the phrase as a backup?
+
+## Future possibilities
+
+[future-possibilities]: #future-possibilities
+
+A significant improvement could be made by using the seed/key generated by the **Secret Phrase** to create a HD Key structure as per [BIP-0032](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) (an implementation of [BIP-044](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) which builds on BIP-0032 already exists in the [JS SDK](https://github.com/casper-ecosystem/casper-js-sdk/blob/master/src/lib/CasperHDKey.ts "CasperHDKey.ts")). The user would only need to manage one secret instead of a secret key file or **Secret Phrase** per keypair.
```

Comment:
You mention using **Secret Phase** in the **Reference-level explanation** section. Does it mean it's no longer _future possibilites_ or am I misunderstanding something?

Reply:
The future possibility is that the key derived from the Secret Phrase could in turn be used to derive many child keys which can be indexed (as per BIP32 / BIP44).

The benefit is having only one secret to manage rather than one per keypair since you only need to recover one key to get all of your keypairs back.

Currently it would be a Secret Phrase per generated Keypair.